validate_tag.sh

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
+#
+# Set of Regular expressions / checks to ensure compliance of git version tags with SODALITE standards
+#
+# <valid prerelease tag> ::= <version core> "-" <pre-release>
+#                          | <version core> "-" <pre-release> "+" <build>
+#                          | <major release>
+#
+# <valid release tag> ::= <version core>
+#                       | <version core> "+" <build>
+#                       | <major release>
+#
+# <major release> ::= "M18Release" | "M24Release" | "M36Release"
+#
+# <version core> ::= <major> "." <minor> "." <patch>
+
+# Usage:
+#   $0 [SemVar|SemVarStage|SemVarProd|MajRel|production|staging] <tag>
+
+FUNC=$1
+VALUE=$2
+
+# Functions
+
+SemVar() {
+  if [[ "$VALUE" =~ ^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(-((0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*))*))?(\+([0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*))?$ ]]; then
+    echo true | tr -d '\n'
+  else
+    echo false | tr -d '\n'
+  fi
+}
+
+SemVarStage() {
+  if [[ "$VALUE" =~ ^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(-((0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z-]*))*))(\+([0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*))?$ ]]; then
+    echo true | tr -d '\n'
+  else
+    echo false | tr -d '\n'
+  fi
+}
+
+SemVarProd() {
+  if [[ "$VALUE" =~ ^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(\+([0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*))?$ ]]; then
+    echo true | tr -d '\n'
+  else
+    echo false | tr -d '\n'
+  fi
+}
+
+MajorRelease() {
+  if [[ "$VALUE" == "M18Release" ]] || [[ "$VALUE" == "M24Release" ]] || [[ "$VALUE" == "M36Release" ]]; then
```

Comment:
M30Release should also be considered as a major release

Reply:
True. This came directly from xOpera-rest-api, so we should probably let them know to make the same change.